assets/src/stories-editor/blocks/amp-story-page-attachment/edit.js

```diff
@@ -0,0 +1,301 @@
+
+		const attachmentClass = classnames( 'amp-page-attachment-content', {
+			'has-text-color': textColor.color,
+			[ textColor.class ]: textColor.class,
+		} );
+		const attachmentStyle = {
+			color: textColor.color,
+			backgroundColor: appliedBackgroundColor,
+		};
+		return {
+			style: attachmentStyle,
+			attachmentClass,
+		};
+	}
+
+	removePost() {
+		this.props.setAttributes( { postId: null } );
+		this.setState( {
+			selectedPost: null,
+			failedToFetch: false,
+		} );
+	}
+
+	render() {
+		const {
+			attributes,
+			setAttributes,
+		} = this.props;
+
+		const {
+			theme,
+			text,
+			title,
+			wrapperStyle,
+			attachmentClass,
+			postId,
+		} = attributes;
+
+		const { selectedPost, searchValue, failedToFetch } = this.state;
+
+		const themeOptions = [
+			{ value: 'light', label: __( 'Light', 'amp' ) },
+			{ value: 'dark', label: __( 'Dark', 'amp' ) },
+		];
+
+		return (
+			<>
+				<InspectorControls>
+					<PanelBody title={ __( 'Page Attachment Settings', 'amp' ) }>
+						<SelectControl
+							label={ __( 'Theme', 'amp' ) }
+							value={ theme }
+							options={ themeOptions }
+							onChange={ ( value ) => {
+								setAttributes( { theme: value } );
+							} }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				{ this.state.isOpen &&
+					<div className={ classnames( 'attachment-container', {
+						'theme-dark': 'dark' === theme,
+					} ) }>
+						<div className="attachment-wrapper">
+							<div className="attachment-header">
+								<span
+									onClick={ () => {
+										this.toggleAttachment( false );
+									} }
+									tabIndex="0"
+									className="amp-story-page-attachment-close-button"
+									role="button"
+									onKeyDown={ () => {
+										// @todo
+									} }
+								/>
+								<RichText
+									value={ title }
+									tagName="span"
+									wrapperClassName="amp-story-page-attachment-title"
+									onChange={ ( value ) => setAttributes( { title: value } ) }
+									placeholder={ __( 'Write Title', 'amp' ) }
+									onClick={ ( event ) => event.stopPropagation() }
+								/>
+								{ postId && (
+									<Button
+										className="remove-attachment-post"
+										onClick={ ( event ) => {
+											event.stopPropagation();
+											this.removePost();
+										} }
+										isLink
+										isDestructive>
+										{ __( 'Remove Post', 'amp' ) }
+									</Button>
+								) }
+							</div>
+							<div className={ attachmentClass } style={ wrapperStyle }>
+								{ selectedPost && selectedPost.content && (
+									<RawHTML>{ selectedPost.content.rendered }</RawHTML>
+								) }
+								{ ( ! postId || failedToFetch ) && (
+									<>
+										{ failedToFetch && (
+											<span>{ __( 'The selected post failed to load, please select a new post', 'amp' ) }</span>
+										) }
+										<PostSelector
+											placeholder={ __( 'Search & select a post to embed content.', 'amp' ) }
```

Comment:
Not sure if there should be a more detailed description when nothing is selected.

Reply:
We can always change it if it doesn't feel right. It's just a text :-)